psi4/src/psi4/libfock/jk.cc

```diff
@@ -179,6 +179,12 @@ std::shared_ptr<JK> JK::build_JK(std::shared_ptr<BasisSet> primary, std::shared_
             return build_JK(primary, auxiliary, options, "DISK_DF");
         }
 
+    } else if (jk_type == "DIRECT_DF_LINK") {
+        CompositeJK* jk = new CompositeJK(primary, auxiliary, "DIRECT_DF", "LINK", options);
+        // Composite JK algorithms should use Incremental Fock build by default
+        if (!options["INCFOCK"].has_changed()) jk->set_incfock(true);
+
+        return std::shared_ptr<JK>(jk);
```

Comment:
A thing I want to bring to the attention of reviewers here, that Andy and I discussed yesterday:
1. This entire formation of the CompositeJK shared pointer could more cleanly be done with a call to std::make_shared. However, the method as implemented was chosen because other JK shared pointers (e.g., DirectJK), were built in this manner - we considered it a matter of convention to do the construction of the CompositeJK shared pointer the way we did. So, my question is, would it fall outside the scope of this PR to use std::make_shared for the construction of the JK shared pointer, and change the other JK shared pointers to do the same?

Reply:
Cleaned it up :)